### PR TITLE
Feature/time types

### LINKED ISSUE #24: Time types
How do we plan on handling time types in this new hal, seems currently we are just using a u32 for frequency etc (perhaps not a bad thing). Historically each hal defined its own time types, but nowadays there are a couple of external crates that will do the job.

* https://crates.io/crates/embedded-time
* https://crates.io/crates/fugit

### PR DESCRIPTION
**Pull request by JurajSadel**
Closes #24

### CODE REVIEW

**Comment by MabezDev** [comment]
To fix the compile errors, although we don't want to touch the utils too much, we'll need to add a conversion from TICKS to NanoSecondsu32. I think you can use https://docs.rs/fugit/0.3.5/fugit/struct.Duration.html#method.from_ticks to do that.

File: esp32-hal/examples/i2c_display.rs [hunk 1]
```diff
@@ -49,7 +50,7 @@ fn main() -> ! {
         peripherals.I2C0,
         io.pins.gpio32,
         io.pins.gpio33,
-        100_000,
+        HertzU32::kHz(100),
```
Comment (MabezDev) [inline esp32-hal/examples/i2c_display.rs:53]: Does the extension trait not work here? Can we do `100.khz()` instead?

**Review by MabezDev — APPROVED** [review]
Looking good from my end, just a small nit about the examples, nothing to worry about though.

File: esp32-hal/examples/i2c_display.rs (hunk 1 above)
Comment (JurajSadel) [reply esp32-hal/examples/i2c_display.rs:53]: Yes, we can. I just wanted to be more persistent (because of `CONST`s).

Comment (jessebraham) [reply esp32-hal/examples/i2c_display.rs:53]: I believe you need to specify the type in order to use the extension trait, but this is already being done in the `spi_loopback` examples: `100u32.kHz()` for example. I would prefer this form be used, personally.

**Pull request merged**